### ISSUE TITLE
Incorrect icons

### ISSUE THREAD

**Issue by dylanparry** [deadalnix/pixel-saver#132]
Hi,

Firstly, an excellent extension---thanks :+1: 

I've been having an issue where the icons used for the close/minimise/window buttons appear to be using the wrong theme when my applications are maximised. Here are some examples:

![window](https://cloud.githubusercontent.com/assets/3692873/26782380/450ea3a4-49eb-11e7-968c-e44c91bdb492.png)

In the above screenshot, the application is in windowed mode. The icons being used for the buttons are the Pop theme, and are correctly displayed.

![maximised](https://cloud.githubusercontent.com/assets/3692873/26782406/64c90450-49eb-11e7-8bc6-a288cb6d160d.png)

In this screenshot, the application is maximised. It correctly shows the title and the buttons on the Gnome top bar, but notice that the buttons aren't the same. They appear to be using the Ambience theme, which is the default for Ubuntu, rather than the theme I have chosen.

Any ideas as to what is causing this, and if so is there anything I can do to fix it?

Thanks,

Dylan

**Comment by dylanparry** [comment]
After reading through the other issues here, I realise that this extension doesn't use the native icons provided by the theme in GTK and have started to look in the themes directory provided by this extension.

I note however that the icons being used are still not correct as the theme I'm using isn't supported by this extension, so it should I presume be falling back on the `default` theme. As the screenshots above show, it's using the `Ambience` theme rather than the default.

I've also tried adding a `Pop` directory to the themes directory and copying everything out of the default theme to begin making my own theme, but despite a reboot, I'm still getting the Ambience icons.

Any idea why it's defaulting to the incorrect theme name? If I can change that, then I'll be able to create a theme for the Pop icons :)

Thanks.

**Comment by dhpowrhost** [comment]
Yes, it would be really nice if the icons would be picked from the GTK+ theme. Just like in Unity 7. That looks great

**Comment by twidxuga** [comment]
+1 on this request. As users are trying to migrate from unity to gnome 3 (now that ubuntu is dropping unity), it is possible that that will impact more users.

**Comment by deadalnix** [comment]
I tried to experiment with using theme icon directly. The long story short is that it break horribly on some themes. If some has a way to make it work nicely I'd be happy as I indeed think it's a better way to do it.

**Comment by rockorequin** [comment]
What does it look like when it breaks horribly? And how do you actually change it to use the theme icons directly (I browsed through the code but couldn't see where to do this)?

Pixel-saver is an excellent extension and this issue would be nice to fix.

**Comment by alvitawa** [comment]
When I installed pixel-saver I had the right icons, but then they changed to the unity ones(I think it happened when I tried to install [global-appmenu](https://github.com/lestcape/Gnome-Global-AppMenu)) What rangiTM suggested didn't work for me, but switching the icon theme to something else and back to the theme I want(Adwaita in my case) got me the right icons back(for now at least).

**Comment by AntonYarchuk** [comment]
+1 for this request.
Another workaround: you should restart gnome.
1. Alt + F2
2. type "r" 
3. Hit Enter

PS: this extension is awesome!!

**Comment by cj-sv** [comment]
Yeah, I'm having the same issue on GNOME Shell 3.24.2 on every restart. Easy fix is to restart gnome-shell. Moved here from the Htitle Firefox Extension which has been continued. Hopefully gets fixed soon. Keep up the good work guys~

**Comment by imyxh** [comment]
Same problem. Is it possible this is only impacting Ubuntu users that switched from Unity to gnome, and then upgraded gnome via ppa?
That's what I did, and the only workaround that works for me is restarting the gnome shell.

**Comment by alvitawa** [comment]
@imyxh I upgraded from ubuntu as well, and I have had the problem, but I've had the right icons now since I posted here 9 days ago.

**Comment by cj-sv** [comment]
@imyxh Probably not since I'm using Ubuntu GNOME.

**Comment by rangiTM** [comment]
I use Ubuntu GNOME too. alt+f2 → r works for me.

**Comment by ebruck** [comment]
Another workaround:

```
cd ~/.local/share/gnome-shell/extensions/pixel-saver@deadalnix.me/themes
mv Ambiance Ambiance.bak
ln -s default Ambiance
```

**Comment by imyxh** [comment]
@ebruck Best workaround so far for me, as I can choose which theme to use and it should persist after a reboot, though I have yet to find out.

**Comment by AntonYarchuk** [comment]
@ebruck Worked fine for me. Many thanks!

**Comment by StrangeNoises** [comment]
The behaviour I see is that, at first it showed the default Adwaita icons, and continued to do so after I switched to using Ambiance. Then I switched to Arc, but Pixel Saver is *now* using the Ambiance icons. (So is it just using the *previous* configuration somehow?) But with a slight twist: If I mouse-over the icons, the mouse-over variants are those of the Arc theme, but with no mouse-over, they're still Ambiance...

Otherwise of course i love what this ext does, especially now I've got them over on the left side and got rid of the Activites button...

(All these themes installed from Ubuntu package management, 17.04; not installed as user themes.)

**Comment by jessedubord** [comment]
+1 on this issue. Encountered using Ubuntu 17.10 (Beta 2) with GNOME shell 3.26.1.

**Comment by StrangeNoises** [comment]
Interesting, the whole extension stopped working for me at all on that platform. Fails to start, with "Error" showing on the extensions web page when I try. I've been waiting for a new release, assuming it wouldn't only be affecting me.

**Comment by roothorick** [comment]
18.04 also impacted. Further, apparently gnome-shell cannot be restarted in Wayland mode, which appears to be the default now.

**Comment by keunes** [comment]
I just did a fresh install of Ubuntu 18.04, and installed the Yaru (Communitheme) snap to upgrade the visuals. As I missed Unity's vertical spacial efficiency, I installed this lovely GNOME Shell extension. Indeed and unfortunately, the theme's icons don't get picked up by the extension (instead displaying Ambience icons).

Would be great if this could be solved.

**Comment by tallero** [comment]
Is still there someone who suffer from this bug?
